src/ray/raylet/worker_pool.cc

```diff
@@ -1269,8 +1258,15 @@ void WorkerPool::PrestartWorkers(const TaskSpecification &task_spec, int64_t bac
 
 bool WorkerPool::DisconnectWorker(const std::shared_ptr<WorkerInterface> &worker,
                                   rpc::WorkerExitType disconnect_type) {
-  runtime_env_manager_.RemoveURIReference(worker->WorkerId().Hex());
   auto &state = GetStateForLanguage(worker->GetLanguage());
+  auto it = state.worker_processes.find(worker->GetStartupToken());
+  if (it != state.worker_processes.end()) {
+    it->second.num_registered_workers--;
```

Comment:
Is it impossible the worker disconnected was a "starting workers"?

Reply:
It looks like this code could be problematic if that's the case

Reply:
Good point. I think this could happen by low probability. I have deleted this field and add a field `alive_started_workers` to record the started workers.